### ISSUE TITLE
Accuracy 0% - finetune() -> division by 0

### ISSUE THREAD

**Issue by timothyvanco** [wangxin0716/ssl#2]
Hi, I tried to run your code in google Colab, but after 1000 epochs I got:

"`Train epoch 1000, SimCLR loss: 5.2841: 100%|██████████| 97/97 [01:02<00:00,  1.56it/s]`"

And then I tried to run finetune() and I an error in "get_lr()" because there were division by 0. So I add there small number:

```
def get_lr(step, total_steps, lr_max, lr_min):
    """Compute learning rate according to cosine annealing schedule."""
    return lr_min + (lr_max - lr_min) * 0.5 * (1 + np.cos(step / (total_steps * np.pi + 0.000000001)))
```

Then I run it again but receive 0.0% accuracy. Please, can you help me with this?

**Comment by p3i0t** [comment]
Hi, I am not sure about your problem. As far as I know, 0% accuracy is quite strange. Even your model is completely untrained, simple blind guessing would give you ~10% accuracy.

**Comment by tarunn2799** [comment]
Hey so adding that value there is a hacky fix, and does not solve your issue. 

The len(dataloader) value returned is simply wrong (its 0). @wangxin0716 can you please explain why and how the subsetrandomsampler works in line 106 of simclr_lin.py?

**Comment by zyxforever** [comment]
> 
It is because the value of len(train_loader) is zero

**Comment by EricZimmermann** [comment]
@tarunn2799 the len of the sampler (100) is less than the batch size (512) . Since drop last is true, it will always drop the batch and has a visible len of 0. @wangxin0716 what batch size do you use at test time?

**Comment by Nandini2801** [comment]
What needs to be done to prevent this issue?
